src/csi/pkg/pos/dagent.go

```diff
@@ -70,29 +70,53 @@ type Status struct {
 	Solution    string `json:"solution,omitempty"`
 }
 
-
 func (dagent *DAgent) ListVolumes(config map[string]string, mtx2 *sync.Mutex) (model.Response, error) {
-    url := fmt.Sprintf("http://%s:%s/api/ibofos/v1/volumelist/%s", config["provisionerIp"], config["provisionerPort"], config["array"])
-    resp, err := util.CallDAgent(url, nil, "GET", "ListVolumes", mtx2)
-    if err != nil {
-        return model.Response{}, status.Error(codes.Unavailable, err.Error())
-    }
-    body, err := ioutil.ReadAll(resp.Body)
-    if err != nil {
-        return model.Response{}, status.Error(codes.Unavailable, err.Error())
-    }
-    dec := json.NewDecoder(bytes.NewBuffer(body))
-    dec.UseNumber()
-    response := model.Response{}
-    if err := dec.Decode(&response); err != nil {
-        return model.Response{}, status.Error(codes.Unavailable, err.Error())
-    }
-    if response.Result.Status.Code == 0 {
-        klog.Info("ListVolumes Success ")
-    } else {
-        return model.Response{}, status.Error(codes.Unavailable, response.Result.Status.Description)
-    }
-    return response, nil
+	url := fmt.Sprintf("http://%s:%s/api/ibofos/v1/volumelist/%s", config["provisionerIp"], config["provisionerPort"], config["array"])
+	resp, err := util.CallDAgent(url, nil, "GET", "ListVolumes", mtx2)
+	if err != nil {
+		return model.Response{}, status.Error(codes.Unavailable, err.Error())
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return model.Response{}, status.Error(codes.Unavailable, err.Error())
+	}
+	dec := json.NewDecoder(bytes.NewBuffer(body))
+	dec.UseNumber()
+	response := model.Response{}
+	if err := dec.Decode(&response); err != nil {
+		return model.Response{}, status.Error(codes.Unavailable, err.Error())
+	}
+	if response.Result.Status.Code == 0 {
+		klog.Info("ListVolumes Success ")
+	} else {
+		return model.Response{}, status.Error(codes.Unavailable, response.Result.Status.Description)
+	}
+	return response, nil
+}
+
+func (dagent *DAgent) VolumeInfo(volName string, config map[string]string, mtx2 *sync.Mutex) (model.Response, error) {
+	url := fmt.Sprintf("http://%s:%s/api/ibofos/v1/array/%s/volume/%s", config["provisionerIp"], config["provisionerPort"], config["array"], volName)
+	resp, err := util.CallDAgent(url, nil, "GET", "VolumeInfo", mtx2)
+	if err != nil {
```

Comment:
Minor comment: I think the lines are too dense. I suggest to have blank lines between code chunks (a block of related blocks).